libs/sdk/src/Util.ts

```diff
@@ -0,0 +1,36 @@
+import { logError } from './Logger';
+
+export function filterNaN(value: Record<string, unknown> | null): Record<string, unknown> | null {
+  if (value == null) return null;
+
+  function convertNestedObjects(obj: Record<string, unknown>): Record<string, unknown> {
+    const updatedObj: Record<string, unknown> = {};
+
+    try {
+      for (const [key, val] of Object.entries(obj)) {
+        if (typeof val === 'number' && isNaN(val)) {
+          // Drop NaN values, as it is not supported by the React Native bridge layer.
+          continue;
+        } else if (val === null) {
+          updatedObj[key] = null;
+        } else if (typeof val === 'object') {
+          if (!Array.isArray(val)) {
+            updatedObj[key] = convertNestedObjects(val as Record<string, unknown>);
+          } else {
+            updatedObj[key] = val.map((item) => {
+              return convertNestedObjects(item as Record<string, unknown>);
```

Comment:
What if `item` is a primitive data type like a string or number?

The recursive logic needs to be fixed.

Reply:
Control is only expected to reach this `else` block when it is an array. 

As per my understanding, primitive types should be handled over [here](https://github.com/rudderlabs/rudder-sdk-react-native/blob/926fe0b7fbf97f41ac44f2d86be529333eed6b31/libs/sdk/src/Util.ts#L25).

Please let me know if I'm missing anything or if I'm expected to handle this case differently.

If you could suggest some edge case value, it would be great.

Reply:
I think I got this. I'll push the respective changes soon. Thanks for pointing.